### ISSUE TITLE
Port twisted.python.context to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5934]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5934|
|Type|enhancement|
|Created|2012-09-02 22:01:14Z|
|Branch|https://github.com/twisted/twisted/tree/context-python3-5934|

[ticket:5932 twisted.python.log depends on this].

<details><summary>Searchable metadata</summary>

```
trac-id__5934 5934
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_context_python3_5934 branches/context-python3-5934
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1346623274000000 1346623274000000
changetime__1346692979000000 1346692979000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [35485]) Branching to 'context-python3-5934'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/context-python3-5934)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @itamarst|
|-|-|

Looks good, I will merge (nice to have better tests).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [35497]) Merge context-python3-5934

Author: exarkun
Review: itamar
Fixes: [#5934](https://github.com/twisted/twisted/issues/5934)

Port twisted.python.context to Python 3.
